code-env/python/spec/requirements.txt

```
@@ -1 +1,14 @@
+spacy==3.0.0
```

Safer not to use any X.0.0 versions. These are usually the first big releases with breaking changes and unforeseen new bugs. To avoid this "bleeding edge" effect, it's better to wait for the subsequent bugfix releases. A good way to do this is:
```suggestion
spacy~=3.0
```

---

It's worth taking the time to read: https://github.com/explosion/spaCy/releases

---

Also, you need to add [th] to get the thai dependency